client/src/document.js

```diff
@@ -157,32 +136,48 @@ function SidebarLeaf({ parent }) {
   );
 }
 
-function SidebarLeaflets({ node }) {
-  const listItems = node.content.map(childNode => {
-    if (childNode.content) {
-      return (
-        <li key={childNode.title}>
-          <SidebarLeaflets node={childNode} />
-        </li>
-      );
-    } else {
-      return (
-        <li
-          key={childNode.uri}
-          className={childNode.isActive ? "active" : undefined}
-        >
-          <Link to={childNode.uri}>{childNode.title}</Link>
-        </li>
-      );
+/**
+ * Walk through the tree, annotating its nodes by adding attributes:
+ * - `isActive` if a node contains the link to the current page
+ * - `open` if a node is on the path to the `isActive` node
+ */
+function setOpenNodes(node, pathname) {
+  for (const child of node.content) {
+    if (child.uri === pathname) {
+      child.open = true;
+      child.isActive = true;
```

Comment:
This I think does not work, because it only sets `open` on the final `details` element in the tree. I want to set `open` on all the `details` elements that are on the path to the active page. So when I navigate to, say http://localhost:3000/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt/asUintN, then I can see the link to that page and its siblings in the sidebar. That's the reason for it returning a value in the original: a node gets "open" if it is the active tree or if one of its children got "open".

Reply:
Using my branch this is what I get:
<img width="483" alt="Screen Shot 2020-03-05 at 2 18 27 PM" src="https://user-images.githubusercontent.com/26739/76017202-3a33fd80-5eec-11ea-8447-d734872842c7.png">
and
<img width="447" alt="Screen Shot 2020-03-05 at 2 19 02 PM" src="https://user-images.githubusercontent.com/26739/76017250-4b7d0a00-5eec-11ea-9f01-b6f887006234.png">

Is that not right?

Reply:
The DOM looks at follows:
<img width="577" alt="Screen Shot 2020-03-05 at 2 19 46 PM" src="https://user-images.githubusercontent.com/26739/76017320-68b1d880-5eec-11ea-95ed-b1962e31148f.png">

Reply:
If I type "yarn start" then open http://localhost:3000/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt/asUintN, I get:

<img width="653" alt="Screen Shot 2020-03-05 at 11 23 49 AM" src="https://user-images.githubusercontent.com/432915/76017651-cbe34100-5ed3-11ea-9497-d7cbd14ba234.png">

Reply:
Weird. Is my browser remembering something for me? Your point makes sense. I'll try to understand this better.

Reply:
> The DOM looks at follows:

Sorry if I'm being thick, but I don't understand how those `details` elements are getting `open` attributes here. As far as I can see, `open` is only set if `node.open` is true (https://github.com/mdn/stumptown-renderer/blob/6818773b049549fe0a118f971e9520405b09e5f2/client/src/document.js#L157), and `node.open` seems to get set only if the node's `uri` matches `pathname` (https://github.com/mdn/stumptown-renderer/blob/6818773b049549fe0a118f971e9520405b09e5f2/client/src/document.js#L147).

So from the screenshot, how is the first `details` under "Reference", which is "Global objects", and doesn't even have a `uri`, getting the `open` attribute?

As I say, sorry if I'm being thick.

Reply:
No, *I'm* being thick because your logic about the algorithm checks out. 

Also, how come it works for me in FirefoxNightly and Chrome 
<img width="712" alt="Screen Shot 2020-03-06 at 7 11 31 AM" src="https://user-images.githubusercontent.com/26739/76082631-d5bf7f80-5f79-11ea-9ebf-1311672c5375.png">

Why do you get something else in your browser?!

Reply:
I put a `console.log()` near where it does `node.open=true;` and, as expected, it only happens once. I'm starting to think the browser fills in the rest. 
By the way, I took the DOM (as outerHTML) from Chrome's devtools and put it straight into Codepen. Still works. 
https://codepen.io/peterbe/pen/jOPaPmZ?editors=1000

Reply:
Here's a strange one for ya. When I run `yarn deployment-run` and open http://localhost:5000/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt/asUintN/ I get this:
<img width="888" alt="Screen Shot 2020-03-06 at 7 44 43 AM" src="https://user-images.githubusercontent.com/26739/76084687-7152ef00-5f7e-11ea-900a-f1e47d4e800f.png">
Which is the same as you got. 
So I downloaded the HTML of that server-rendererd page.
(which is the same as `stumptown-renderer/client/build/en-us/docs/web/javascript/reference/global_objects/bigint/asuintn/index.html`) and I format (pretty print it) and it comes out like this:
<img width="1616" alt="Screen Shot 2020-03-06 at 7 47 00 AM" src="https://user-images.githubusercontent.com/26739/76084910-f2aa8180-5f7e-11ea-92d0-ed359e2939dd.png">

Now it doesn't even have the `open=""` thing for anything! 
But since I have my `console.log()` still in there for when it does `node.open=true;` I still saw that happening so it could be `cheerio` that cleans it up like that.

Reply:
https://codesandbox.io/s/tender-antonelli-21z8f demonstrates the bug I introduced. I.e. I only set `node.open` on the absolute botton leaf node. 
Clearly, this is what the expected outcome is with an algorithm bug like mine. But when does it not happen in stumptown-renderer????

Reply:
Forked, now it sets `open:true` on all the nodes parents: https://codesandbox.io/s/priceless-snowflake-9ovlq
that clearly fixes the bug. 
Still doesn't explain why it's working in stumptown-renderer. But I'm gathering my sanity.

Reply:
Now I've finally made a smidge of progress. Turns out, when I built the index.json file it got `open` set in there. That's because we didn't properly clone. Instead, we mutated the object in the React land and when the ssr component had rendered it it produces a `index.json` and an `index.html`. So no wonder it was weird when I didn't see anything that made sense on `localhost:3000`. 
<img width="946" alt="Screen Shot 2020-03-06 at 10 26 56 AM" src="https://user-images.githubusercontent.com/26739/76097094-18428580-5f95-11ea-9d90-faa00029a7cc.png">
Yet another lesson learned to never mutate state.